src/l3/ip.rs

```diff
@@ -171,3 +154,37 @@ impl Ipv6Hdr {
         bitfield_unit
     }
 }
+
+/// Protocol which is encapsulated in the IPv4 packet.
+/// <https://en.wikipedia.org/wiki/List_of_IP_protocol_numbers>
+#[repr(u8)]
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+pub enum IpProto {
```

Comment:
Is there a reason this enum is incomplete?

Reply:
my guess is that more PRs like https://github.com/vadorovsky/network-types/pull/2/files are still needed to complete it 😄

Reply:
No reason apart from my laziness. I will add all protocols during next push.

Reply:
done